src/edu/csus/ecs/pc2/shadow/ShadowController.java

```diff
@@ -690,11 +692,33 @@ private Map<String, ShadowJudgementInfo> getJudgementsMap(Run[] pc2Runs) {
                         } else {
                             // we've exhausted methods of obtaining an acronym
                             log.warning("Null judgement acronym for run " + run.getNumber() + ", judgement string " + judgementString + "; skipping");
+
+                            Judgement judgement = localContest.getJudgement(jr.getJudgementId());
+                            if (judgement == null) {
+                                log.warning("Null judgement acronym for run " + run.getNumber() + ", Judgement not found in model for run");
```

Comment:
s/judgement acronym/judgment/

Reply:
This added code block is inside the `else` for the condition  `(jr != null)`.  Therefore, this code will get executed when (and only when) `jr == null`.  But this added code calls `jr.getJudgementId()`.  I believe this code won't do what is intended; it will throw an NPE.

Cannot approve this change.

Reply:
I agree; it's misleading to say the judgement ACRONYM is null; it's the Judgement that's null at this point.